Trivial change arf

ezyang opened the pull request:
Trivial change arf

Summary: none

Differential Revision: https://phabricator.intern.facebook.com/D9514646

ezyang commented on torch/csrc/jit/ir.h:
```diff
@@ -226,46 +233,75 @@ TH_DISALLOW_COPY_AND_ASSIGN(Graph);
+  addInput(graph_->value_traces.at(value));
+}
+
+inline Node & Node::registerOutput(value_type * output) {
```
So, the naming convention is, add* doesn't return, but register* does (for chaining)?

apaszke replied:
I felt that `register` is a better name, because it doesn't really modify the IR in any way - it doesn't increase any output counts or anything; it only registers the mapping. But I don't have a strong preference and can change it if you want

zdevito replied:
This will be much clearer if TraceState handles the mappings while Graph represents the computation.

ezyang commented on torch/csrc/jit/python_tracer.cpp:
```diff
@@ -48,9 +48,12 @@ PyObject * THPTracer_exit(PyObject *_unused, PyObject *args)
-    graph->addOutput(((THPVariable*)output_obj)->cdata->trace_value);
+    try {
+      graph->addOutput(((THPVariable*)output_obj)->cdata.get());
+    } catch (std::logic_error& e) {
```
I wonder when we're supposed to use logic_error as opposed to Python error reporting mechanism.

apaszke replied:
I think a rule of thumb would be that all files use C++ exceptions, except for those prefixed with `python_` that use Python error reporting

ezyang commented:
On a cursory review looks fine to me.

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -11,12 +11,15 @@
 
 namespace torch { namespace jit {
 
+using value_type = torch::autograd::Variable;
```
I strongly recommend we avoid allowing the IR itself from referring directly to values. It should express computations alone. If we really need to represent a Tensor, we can add a Constant Node, but even then it should not be a Variable object.

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -226,46 +233,75 @@ TH_DISALLOW_COPY_AND_ASSIGN(Graph);
-  }
+
+  // maintains a mapping from Variables to Nodes that produced them
+  std::unordered_map<value_type*, Node*> value_traces;
```
This should be in the TraceState, not in the Graph. This is what is forcing the coupling between Variable and Node in the IR

ezyang replied:
Oops, I missed this! Absolutely yes.

zdevito commented on torch/csrc/jit/ir.h:
```diff
@@ -226,46 +233,75 @@ TH_DISALLOW_COPY_AND_ASSIGN(Graph);
+    output_ = create<Return>();
+  }
+
+  void addInput(value_type * value) {
```
Again, can be defined on TraceState.

apaszke commented:
Yeah, moving it to the tracing state sounds better. I'll change it tomorrow

ezyang commented:
When CI is green, ship it!!

ezyang committed:
Trivial change arf

Summary: none

Differential Revision: https://phabricator.intern.facebook.com/D9514646
